app/actions/websocket/index.ts

```diff
@@ -64,90 +68,119 @@ export async function handleClose(serverUrl: string, lastDisconnect: number) {
         }
+    }
+
+    let removeTeams;
+    if (removeTeamIds?.length) {
+        // Immediately delete myTeams so that the UI renders only teams the user is a member of.
+        removeTeams = await queryTeamsById(database, removeTeamIds);
+        await deleteMyTeams(operator, removeTeams!);
+    }
+
+    let removeChannels;
+    if (removeChannelIds?.length) {
+        removeChannels = await queryChannelsById(database, removeChannelIds);
+    }
+
+    const modelPromises = await prepareModels({operator, initialTeamId, removeTeams, removeChannels, teamData, chData, prefData, meData});
+    if (rolesData.roles?.length) {
+        modelPromises.push(operator.handleRole({roles: rolesData.roles, prepareRecordsOnly: true}));
+    }
 
-        channelMemberships = memberships;
-
-        if (currentChannelId) {
-            const stillMemberOfCurrentChannel = memberships?.find((cm) => cm.channel_id === currentChannelId);
-            const channelStillExist = channels?.find((c) => c.id === currentChannelId);
-            const viewArchivedChannels = config?.ExperimentalViewArchivedChannels === 'true';
-
-            if (!stillMemberOfCurrentChannel) {
-                handleUserRemovedFromChannelEvent(serverUrl, {data: {user_id: currentUserId, channel_id: currentChannelId}});
-            } else if (!channelStillExist ||
-                (!viewArchivedChannels && channelStillExist.delete_at !== 0)
-            ) {
-                handleChannelDeletedEvent(serverUrl, {data: {user_id: currentUserId, channel_id: currentChannelId}} as WebSocketMessage);
-            } else {
-                // TODO Differentiate between post and thread, to fetch the thread posts
-                fetchPostsSince(serverUrl, currentChannelId, lastDisconnectedAt);
+    if (modelPromises.length) {
+        const models = await Promise.all(modelPromises);
+        const flattenedModels = models.flat();
+        if (flattenedModels?.length > 0) {
+            try {
+                await operator.batchRecords(flattenedModels);
+            } catch {
+                // eslint-disable-next-line no-console
+                console.log('FAILED TO BATCH WS reconnection');
             }
         }
+    }
+
+    const currentChannelId = await queryCurrentChannelId(database);
+    if (currentChannelId) {
+        // TODO Differentiate between post and thread, to fetch the thread posts
+        fetchPostsSince(serverUrl, currentChannelId, lastDisconnectedAt);
 
         // TODO Consider global thread screen to update global threads
-    } else if (!teamMembershipsError) {
-        handleLeaveTeamEvent(serverUrl, {data: {user_id: currentUserId, team_id: currentTeamId}} as WebSocketMessage);
+
+        // defer fetching posts for unread channels on initial team
+        if (chData?.channels && chData.memberships) {
+            fetchPostsForUnreadChannels(serverUrl, chData.channels, chData.memberships, currentChannelId);
```

Comment:
What is the difference between calling and not calling this here, having in mind we call something similar in line 177? (Maybe they are not so similar as I think)

Reply:
the idea is that once the WS reconnects, we are fetching the posts for the unread channels that belong to this team which are the ones that `chData` holds, then the other one in to fetch the other teams channels to sync the other teams and then it will fetch the posts for unread channels in those teams, essentially synching the teams / channels / posts in unread channels